### DISCUSSION TITLE
Prompt ResourcePack

### DISCUSSION THREAD

**Discussion by Flamgop** [Minestom/Minestom#971] [Help] [answered]
how

**Comment by GreatWyrm** [comment, marked as answer]
`ResourcePack pack = ResourcePack.optional(url, sha1hash);`
```
MinecraftServer.getGlobalEventHandler().addListener(
    PlayerLoginEvent.class, playerLoginEvent -> playerLoginEvent.getPlayer().setResourcePack(pack)
);
```

**Reply by Flamgop** [reply]
it's not prompting for me for some reason, any idea why?

**Reply by Flamgop** [reply]
oh wait, it does work, but it doesn't prompt for more than a second, and only if you have the resource pack set to forced

**Reply by Flamgop** [reply]
fixed it by sending the packet myself

**Reply by DearFox** [reply]
I would like to add an answer. How I made the window with the choice to download the resource pack not disappear:

`GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();`
...
`globalEventHandler.addListener(PlayerSpawnEvent.class, event -> {
            event.getPlayer().setResourcePack(pack);
        });`

I used the PlayerSpawnEvent event.
As I understand it, it is called when the player appears in the world and not during the loading process.
At least it seems to work.
